src/Projects Page/ProjectsPage.tsx

```diff
@@ -1,49 +1,60 @@
 import * as React from 'react';
+import { connect } from 'react-redux';
+import LoggedInHeader from '../LoggedInHeader';
 import Header from '../Header';
 import Footer from '../Footer';
 import Projects from '../Projects';
 import ProjectsFilter from './ProjectsFilter';
 import '../styles/ProjectsPage.css';
 import '../styles/Project.css';
-import { PassedProps, State, Props } from '../types/ProjectsPage.d';
+import { PassedProps, State } from '../types/ProjectsPage.d';
+import { Store } from '../types/Redux';
 // import { connect } from 'react-redux';
 class ProjectsPage extends React.Component<PassedProps, State> {
-    constructor(props: Props) {
-        super(props);
-        this.state = {
-            searchTerm: '',
-        };
-    }
+  constructor(props: PassedProps) {
+    super(props);
+    this.state = {
+      searchTerm: ''
+    };
+  }
+
+  public onFormChange(e: React.FormEvent<HTMLButtonElement>): void {
+    this.setState({
+      [e.currentTarget.name]: e.currentTarget.value
```

Comment:
we found out this method of setting name and value doesn't work.
use this as an example (also found in login/register.tsx)

```
handleChange = (e: React.FormEvent<HTMLInputElement>): void => {
    var { name, value } = e.currentTarget;

    this.setState({
      [name]: value
      // tslint:disable-next-line
    } as any);
  };
```